### DISCUSSION TITLE
Apex Domain Issued Invalid Certificate with GitHub Pages Custom Domain

### DISCUSSION THREAD

**Discussion by rajsite** [community/community#23184] [Other Feature Feedback, Questions, & Ideas]
<p>With GitHub Pages configured for a custom domain of <code>www.webvi.io</code>:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/b/aba9c3a908c99987764b48efb9a134b044661ea3.png" data-download-href="/uploads/short-url/ouBmG35ZDkpgGuMRpPCI8BZpzfd.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1588923/181075484-0d163ed5-d79e-41c7-8ee9-ee8096f23c1b.png" alt="image" data-base62-sha1="ouBmG35ZDkpgGuMRpPCI8BZpzfd" width="690" height="391" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/b/aba9c3a908c99987764b48efb9a134b044661ea3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">930×527 23.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I am able to successfully access my GitHub pages site from <a href="http://www.webvi.io" rel="noopener nofollow ugc">http://www.webvi.io</a> and <a href="https://www.webvi.io" rel="noopener nofollow ugc">https://www.webvi.io</a>.</p>
<p>I also configured an ANAME record to the FQDN <a href="http://www.webvi.io" rel="noopener nofollow ugc">www.webvi.io</a>:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/c/ccc2edb435425a885a6d9ac481c66ff4a1eef6fb.png" data-download-href="/uploads/short-url/tdp2g0Fgr7wWg7q6efrmuslbWxt.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1588923/181075485-fe6e18db-02de-4f56-8dc5-d12e475d512c.png" alt="image" data-base62-sha1="tdp2g0Fgr7wWg7q6efrmuslbWxt" width="493" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/c/ccc2edb435425a885a6d9ac481c66ff4a1eef6fb_2_493x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/c/ccc2edb435425a885a6d9ac481c66ff4a1eef6fb_2_739x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/c/ccc2edb435425a885a6d9ac481c66ff4a1eef6fb.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/c/ccc2edb435425a885a6d9ac481c66ff4a1eef6fb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">930×943 145 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Which based on the <a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain" rel="noopener nofollow ugc">documentation</a> reproduced as follows:</p>
<blockquote>
<p>For example, if you configure <code>www.example.com</code> as your custom domain for your site, and you have <code>ALIAS</code> and <code>CNAME</code> records set up for the apex and <code>www</code> domains, then <code>example.com</code> will redirect to <code>www.example.com</code> .</p>
</blockquote>
<p>I find that the address <a href="http://webvi.io" rel="noopener nofollow ugc">http://webvi.io</a> is correctly redirected, however the address <a href="https://webvi.io" rel="noopener nofollow ugc">https://webvi.io</a> fails to load and redirect due to a certificate error:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/5/65bba6155d25daef8247ca69ef773e7df499c787.png" data-download-href="/uploads/short-url/evYg7XV4emj3WL6Uxz7TEDKszCD.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1588923/181075486-aba975aa-2947-40f8-a785-7d850b4954c0.png" alt="image" data-base62-sha1="evYg7XV4emj3WL6Uxz7TEDKszCD" width="690" height="457" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/5/65bba6155d25daef8247ca69ef773e7df499c787_2_690x457.png, https://aws1.discourse-cdn.com/github/optimized/3X/6/5/65bba6155d25daef8247ca69ef773e7df499c787_2_1035x685.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/6/5/65bba6155d25daef8247ca69ef773e7df499c787.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/5/65bba6155d25daef8247ca69ef773e7df499c787_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1111×737 179 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>If I manually ignore the security error and continue then the redirect will occur to the <a href="https://www.webvi.io" rel="noopener nofollow ugc">https://www.webvi.io</a> address.</p>
<p>It looks like there was a <a href="https://github.community/t/https-www-dont-tou-ch-https-dont-tou-ch-works-but-not-the-reverse/10885">similar reported issue</a>. Is there some configuration needed to have the <a href="https://webvi.io" rel="noopener nofollow ugc">https://webvi.io</a> correctly forward to the www?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/rajsite">@rajsite</a>, GitHub Pages only generates an SSL certificate for the exact domain you enter in the custom domain input box. This means that only <code> www.webvi.io</code> will be secured. The apex domain <code>webvi.io</code> won’t be.</p>
<p>This means that you’ll get a security error when you visit the HTTPS variant of your apex domain <a href="https://webvi.io/">https://webvi.io/</a>. You can still bypass this security error and you’ll be redirected to the secure version of your <code>www</code> subdomain, but the error will still show. This isn’t an issue if you visit the <em>HTTP</em> variant of your apex domain, <a href="http://webvi.io/">http://webvi.io/</a>, you won’t see a security error here and will be redirected normally.</p>
<p>There isn’t a direct workaround for this yet. If you <em>need</em> to have both versions of your domain secured then you will likely need to use an external service that can generate SSL certificates for you and use that in place of GitHub’s certificate generation. Many people have success using Cloudflare for this.</p>
<p>The Pages team are aware of it though, and are looking to change this at some point in the future, though I can’t give an estimate on when this will happen as it’s not currently on our <a href="https://github.com/github/roadmap/projects/1">public roadmap</a>.</p>

**Comment by rajsite** [comment]
<p>Hi <a class="mention" href="/thomasshaped">@thomasshaped</a>,</p>
<p>If you look at the link to the <a href="https://github.community/t/https-www-dont-tou-ch-https-dont-tou-ch-works-but-not-the-reverse/10885">similar reported issue</a> this does not seem to be the case in the reverse direction.</p>
<p>The existing github project has a <a href="https://github.com/dont-touch-my-mushroom/dont-touch-my-mushroom.github.io/blob/9e7dd3c3da676af6374fa0e1853a13fff6d038d5/CNAME" rel="noopener nofollow ugc">custom domain configured for the APEX</a>. Then they configured A records to point to GitHub Pages servers and a CNAME www to their github account <code>dont-touch-my-mushroom.github.io.</code>.</p>
<p>From that you can see GitHub Pages issues a certificate for the apex and the www:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/a/7a29d231bdee8fea96fe6fe94dfbb72dae0319e6.png" data-download-href="/uploads/short-url/hqHQRuWzXBYsRB07yzEwR34cKyO.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1588923/181075492-54e53330-007a-4e23-b733-b78afcff64da.png" alt="image" data-base62-sha1="hqHQRuWzXBYsRB07yzEwR34cKyO" width="544" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/a/7a29d231bdee8fea96fe6fe94dfbb72dae0319e6_2_544x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/a/7a29d231bdee8fea96fe6fe94dfbb72dae0319e6_2_816x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/7/a/7a29d231bdee8fea96fe6fe94dfbb72dae0319e6.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/a/7a29d231bdee8fea96fe6fe94dfbb72dae0319e6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">874×802 96.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/f/cf84023166c3642228028b54fd8ed686dad9e2b3.png" data-download-href="/uploads/short-url/tBLF8eomUtkqSu6YYX0h9YZbF9F.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1588923/181075493-577f295b-c1e8-411f-aa1f-6e1dac23d9a9.png" alt="image" data-base62-sha1="tBLF8eomUtkqSu6YYX0h9YZbF9F" width="535" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cf84023166c3642228028b54fd8ed686dad9e2b3_2_535x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cf84023166c3642228028b54fd8ed686dad9e2b3_2_802x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/f/cf84023166c3642228028b54fd8ed686dad9e2b3.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/f/cf84023166c3642228028b54fd8ed686dad9e2b3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">863×806 98.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><p></p>
<p>So is it intended that GitHub Pages will only issue certs for the APEX and the WWW when the custom domain for a project is configured for the APEX?</p>
<p>If this is intended then it seems like the <a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain" rel="noopener nofollow ugc">following documentation</a> needs a note to emphasize this is an unsupported configuration:</p>
<blockquote>
<p>For example, if you configure <a href="http://www.example.com" rel="noopener nofollow ugc">www.example.com</a> as your custom domain for your site, and you have ALIAS and CNAME records set up for the apex and www domains, then <a href="http://example.com" rel="noopener nofollow ugc">example.com</a> will redirect to <a href="http://www.example.com" rel="noopener nofollow ugc">www.example.com</a>. Note HTTPS configurations are not supported when using GitHub Pages to redirect from <a href="http://example.com" rel="noopener nofollow ugc">example.com</a> to <a href="http://www.example.com" rel="noopener nofollow ugc">www.example.com</a>. HTTPS is only supported when redirecting from <a href="http://www.example.com" rel="noopener nofollow ugc">www.example.com</a> to <a href="http://example.com" rel="noopener nofollow ugc">example.com</a></p>
</blockquote>
<p>Am I understanding correctly?</p>

**Comment by otter-computer** [comment]
<aside class="quote no-group" data-username="rajsite" data-post="3" data-topic="150184">
<div class="title">
<div class="quote-controls"></div>
 rajsite:</div>
<blockquote>
<p>So is it intended that GitHub Pages will only issue certs for the APEX and the WWW when the custom domain for a project is configured for the APEX?</p>
</blockquote>
</aside>
<p>No, GitHub directly doesn’t issue certs for both in either instance. In the instance you mention there is a valid certificate for both, but the certificate for <code>www.dont-tou.ch</code> will expire and not be renewed after the 77 days.</p>
<p>You can “trick” Pages into generating a certificate for both by changing your domain to the <code>www</code> variant, waiting for a certificate to be issued, then changing it back to the apex, but this second certificate won’t be renewed automatically at the end of the 90 day lifespan of the certificate.</p>

**Comment by rajsite** [comment]
<p>Thanks for the clarification, I think I have a better understanding of the current expected behavior.</p>
<p>For future users I think a temporary note in the documentation would be helpful that describes how users may not see expected behavior with HTTPS enabled when leveraging the GitHub Pages HTTP forwarding feature between a www CNAME and APEX or vice-versa.</p>
<p>I appreciate your help!</p>

**Comment by otter-computer** [comment]
<p>Yeah it is a bit confusing when this happens. For what it’s worth though I believe this will be changed at <em>some point</em> in the future—it’s requested constantly!—but I can’t give you a date on when :sweat_smile:</p>

**Comment by lhoupert** [comment]
<p>Yes I have also the same problem, it would be great if you could implement it at some point!</p>

**Comment by lhoupert** [comment]
<p>The solution I found was to not use the www CNAME entry to the github website but instead redirect the www of my domain to my webhosting IP address.</p>
<p>This way I could generate a SSL certificate with my web hosting provider for my www domain.</p>
<p>Then I edited the .htaccess on my website to add redirection conditions from my www domain to my non-www domain.</p>
<p><strong>But to do that you need a web hosting service (I am using OVH)</strong></p>

**Comment by otter-computer** [comment]
<aside class="quote no-group" data-username="lhoupert" data-post="8" data-topic="150184">
<div class="title">
<div class="quote-controls"></div>
 lhoupert:</div>
<blockquote>
<p>This way I could generate a SSL certificate with my web hosting provider for my www domain.</p>
</blockquote>
</aside>
<p>If your domain provider supports this it is also a great solution, yeah! Many people host their DNS with Cloudflare and they provide SSL certs that cover both domains, so that may be an option for anyone else stumbling upon this thread.</p>
<p>Keep in mind that if you do this though the HTTPS checkbox in your GitHub Pages settings will likely show that your site isn’t secure—even when it is—because GitHub isn’t aware of the SSL certificate generated by an external service.</p>

**Comment by lhoupert** [comment]
<p>I had checked the https checkbox on my github page settings.</p>
<p>When checking the ssl certificate I got the results below:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/e/8e7c04cc1294d78dd3e68a4a03e08a7a9896d74f.png" data-download-href="/uploads/short-url/kktvBo4TQWHPN2ad30hzupRu7VB.png?dl=1" title="Screenshot 2021-01-28 at 14.49.08" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/10154151/181075496-5f8d684a-6cd1-4049-b702-ea4a37308dcb.png" alt="Screenshot 2021-01-28 at 14.49.08" data-base62-sha1="kktvBo4TQWHPN2ad30hzupRu7VB" width="618" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/e/8e7c04cc1294d78dd3e68a4a03e08a7a9896d74f_2_618x500.png, https://aws1.discourse-cdn.com/github/original/3X/8/e/8e7c04cc1294d78dd3e68a4a03e08a7a9896d74f.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/8/e/8e7c04cc1294d78dd3e68a4a03e08a7a9896d74f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/e/8e7c04cc1294d78dd3e68a4a03e08a7a9896d74f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-01-28 at 14.49.08</span><span class="informations">816×660 111 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/f/aff47c5c1f8b73519eb7e25c4ed862b9a321e748.png" data-download-href="/uploads/short-url/p6zmFK3vRCEvHyaorV9QLtre0Zy.png?dl=1" title="Screenshot 2021-01-28 at 14.49.18" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/10154151/181075497-aa7c96d8-bce9-44c0-83e7-0f2cc7fa79f6.png" alt="Screenshot 2021-01-28 at 14.49.18" data-base62-sha1="p6zmFK3vRCEvHyaorV9QLtre0Zy" width="599" height="499" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/f/aff47c5c1f8b73519eb7e25c4ed862b9a321e748_2_599x499.png, https://aws1.discourse-cdn.com/github/original/3X/a/f/aff47c5c1f8b73519eb7e25c4ed862b9a321e748.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/f/aff47c5c1f8b73519eb7e25c4ed862b9a321e748.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/f/aff47c5c1f8b73519eb7e25c4ed862b9a321e748_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-01-28 at 14.49.18</span><span class="informations">798×666 113 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Am I right to think that this is ok?</p>

**Comment by otter-computer** [comment]
<p>In your case it shouldn’t matter really, so you should be fine. If your certificate is controlled by your domain provider that cert should continue to be valid.</p>
<p>If GitHub had previously generated a certificate for your domain then this will just expire and won’t be renewed, but it shouldn’t cause any problems with your site’s actual SSL certificate generated by your domain provider.</p>
<p>Once this GitHub certificate does expire the checkbox in settings may uncheck itself automatically, or even show an error, but your site should continue to work.</p>
